airflow/www/security.py

```diff
@@ -372,13 +378,12 @@ def merge_pv(perm, view_menu):
         dag_role = [role for role in all_roles if role.name not in EXISTING_ROLES]
```

Comment:
`EXISTING_ROLES` is a static set, in which we have 'Admin',  'Viewer', 'User', 'Op', and 'Public'. It's possible that we have created a new role, say `EnhancedUser` for which we added Connection Edit access.

Then this custom role (non-dag specific role) will also be listed into `dag_role` here as it's not in `EXISTING_ROLES` (which should not happen).

But I'm not sure whether this will cause any actual impact.

Reply:
It shouldn't have any impact. The code is to allow all the other non-static roles to have all the User role's permisssion on all User's views except the "all-dag" view. This is to allow the user to access basic functionality in webserver. If we would like to modify and control this part, we could modify later, but it is not part of this pr's scope.